### ISSUE TITLE
Does this library support downloading using the content protocol scheme on Android?

### ISSUE THREAD

**Issue by mitchdowney** [kesha-antonov/react-native-background-downloader#39]
Congrats on the 3.0.0 release! I really appreciate the work you've put into this.

I am trying to get a custom download destination to work with `@kesha-antonov/react-native-background-downloader` and `react-native-scoped-storage`, but when I try to download files using RNBD into the `content://` scheme path that `const dir = await ScopedStorage.openDocumentTree(true)` gives me (I'm reading `dir?.uri`), RNBD just hangs after the `progress` event fires one time. No error seems to be thrown...unless I'm not catching it properly.

Sorry I'm not much of an Android dev. Do you happen to know if I am using the wrong type of destination path with RNBD?

My `download` call looks like this:

```
const task = downloader
  .download({
    "destination": "content://com.android.externalstorage.documents/tree/primary%3APodcasts/document/primary%3APodcasts/OscxaAT15.mp3",
    "headers": {
        "user-agent": "Podverse/Android F-Droid Mobile App"
    },
    "id": "OscxaAT15",
    "url": "https://op3.dev/e,pg=9b024349-ccf0-5f69-a609-6b82873eab3c/podnews.net/audio/podnews240111.mp3?_from=podverse.fm"
})
  ...
```

Does that look correct to you? Or am I misunderstanding how `content://` should be used?

---

Separate issue. The `.begin` callback never seems to be called for me with 3.0.0. Not sure if a bug or I'm doing something wrong. Can create a new bug ticket if that's a bug.

**Comment by kesha-antonov** [comment]
Hi
Thanks!

I'll check those issues in the upcoming days

**Comment by kesha-antonov** [comment]
> Separate issue. The .begin callback never seems to be called for me with 3.0.0. Not sure if a bug or I'm doing something wrong. Can create a new bug ticket if that's a bug.

You're right, that was a bug. Fixed https://github.com/kesha-antonov/react-native-background-downloader/commit/b9e9a4c3d28f4901e70129a9b46a63a265eddca0

**Comment by kesha-antonov** [comment]
I downloads my files into `RNBU.fs.dirs.DocumentDir/...` from https://github.com/RonRadtke/react-native-blob-util/wiki/File-System-Access-API#dirs

**Comment by kesha-antonov** [comment]
You can also try `directories.documents` from this lib: https://github.com/kesha-antonov/react-native-background-downloader?tab=readme-ov-file#downloading-a-file

**Comment by mitchdowney** [comment]
I'm very sorry for the delay! I'm finally back working on my podcast app, and it appears the bug is indeed fixed!

Something that is confusing / surprising to me is that it I *thought* I was going to need to pass in the path that the user specified (which begins with `content://`) as the destination for RNBD...but it appears that `downloader.directories.documents` will work for both default and custom user-specified storage location. I guess somehow behind the scenes RNBD is aware (or maybe it is OS level) that a custom storage location has been specified, and `downloader.directories.documents` is smart enough to use it instead of the default location.

Thanks so much again for the quick fix. If it is working like I think it is then this will be a huge improvement for our users.

**Comment by mitchdowney** [comment]
eek...maybe I don't understand how it's supposed to work after all. Even though I see the files going in the Files > Internal Storage > Podcasts directory...when I press play, it appears that the file actually loads from `file:///storage/emulated/0/Android/data/com.podverse/files/dYIrgmYig.mp3`, which doesn't seem right, because when I delete the file from Files > Internal Storage > Podcasts, then open the app without internet access, the file still plays.

I guess...the file is being downloaded into 2 locations? It is going into the Internal Storage > Podcasts directory when I download, but it seems like the app isn't actually using that, and still manages to find the file through `file:///storage/emulated`. Unless the app is able to find files that within the trash, I'm not sure how to explain this.

Anyway, I will investigate further. Sorry to be trouble.

**Comment by kesha-antonov** [comment]
Hi
Can you give me example of path where you want to download the file and why? What's your use case?

In our use case we want to download file to app's directory and file be not visible to user in file inspector.

**Comment by kesha-antonov** [comment]
> I guess...the file is being downloaded into 2 locations? It is going into the Internal Storage > Podcasts directory when I download, but it seems like the app isn't actually using that, and still manages to find the file through `file:///storage/emulated`. Unless the app is able to find files that within the trash, I'm not sure how to explain this.

File downloaded to https://developer.android.com/reference/android/app/DownloadManager.Request#setDestinationInExternalFilesDir(android.content.Context,%20java.lang.String,%20java.lang.String)

then it gets copied to your path at `destination` param

There's a bug - I forgot to delete source file

**Comment by kesha-antonov** [comment]
> > I guess...the file is being downloaded into 2 locations? It is going into the Internal Storage > Podcasts directory when I download, but it seems like the app isn't actually using that, and still manages to find the file through `file:///storage/emulated`. Unless the app is able to find files that within the trash, I'm not sure how to explain this.
> 
> File downloads to https://developer.android.com/reference/android/app/DownloadManager.Request#setDestinationInExternalFilesDir(android.content.Context,%20java.lang.String,%20java.lang.String)
> 
> then it gets copied to your path at `destination` param
> 
> Also, there's a bug - I forgot to delete source file

Fixed - src file removed after moving to destination

**Comment by kesha-antonov** [comment]
Released https://github.com/kesha-antonov/react-native-background-downloader/releases/tag/3.1.1

**Comment by kesha-antonov** [comment]
@mitchdowney Also look at the comment here https://github.com/podverse/podverse-rn/commit/e2e644017fc73850348e6c2a16965b6430ffcdbb
